nimble/host/src/ble_gap.c

```diff
@@ -1336,14 +1336,36 @@ ble_gap_rx_adv_set_terminated(struct hci_le_adv_set_terminated *evt)
 /* Periodic adv events */
 #if MYNEWT_VAL(BLE_EXT_ADV) && MYNEWT_VAL(BLE_PERIODIC_ADV)
 void
-ble_gap_rx_per_adv_sync_estab(struct hci_le_subev_per_adv_sync_estab *evt)
+ble_gap_rx_peroidic_adv_sync_estab(
+		struct hci_le_subev_periodic_adv_sync_estab *evt)
 {
 	struct ble_gap_event event;
 	struct ble_gap_master_state state;
+	struct ble_hs_periodic_disc * pdisc;
 
 	memset(&event, 0, sizeof event);
 
-	event.type = BLE_GAP_EVENT_PER_ADV_SYNC_ESTAB;
+	/* There must be memory for pdisc as the check was done when creating
+	 * sync
+	 */
+	pdisc = ble_hs_periodic_disc_alloc(evt->sync_handle);
+	BLE_HS_DBG_ASSERT(pdisc != NULL);
+
+	pdisc->sync_handle 		         = evt->sync_handle;
+	pdisc->adv_sid 				     = evt->sid;
+	pdisc->advertiser_addr_type      = evt->adv_addr_type;
+
+	memcpy(pdisc->advertiser_addr.val, evt->adv_addr, 6);
+
+	pdisc->advertiser_phy 			 = evt->adv_phy;
+	pdisc->periodic_adv_itvl 		 = evt->per_adv_ival;
+	pdisc->advertiser_clock_accuracy = evt->adv_clk_accuracy;
+
+	ble_hs_lock();
+	ble_hs_periodic_disc_insert(pdisc);
```

Comment:
btw - please check `ble_hs_stop` where we disconnect all the connections on stack close. We need to add there code t terminate all periodic_sync